src/components/RuleCard/index.jsx

```diff
@@ -189,12 +207,18 @@ function RuleCard({ rule, onRuleDelete, ...props }) {
               </Fragment>
             )
           }
-          titleTypographyProps={{
-            component: 'h2',
-            variant: 'h6',
-          }}
           title={
-            rule.channel ? `${rule.product} : ${rule.channel}` : rule.product
+            <Typography component="h2" variant="h6">
+              {rule.channel
+                ? `${rule.product} : ${rule.channel}`
+                : rule.product}{' '}
+              <a
+                href={`#${rule.rule_id}`}
```

Comment:
Similar to what I discovered in #73, what do we do for pending Rules, which have no rule id yet?

Reply:
I think we should be able to take care of that in https://github.com/mozilla-frontend-infra/balrog-ui/issues/71 where we would be linking to `sc_id`.